ipatests/test_integration/test_commands.py

```diff
@@ -699,3 +699,55 @@ def is_tls_version_enabled(tls_version):
         assert not is_tls_version_enabled('tls1')
         assert not is_tls_version_enabled('tls1_1')
         assert is_tls_version_enabled('tls1_2')
+
+    def test_sss_ssh_authorizedkeys(self):
+        """Login via Ssh using private-key for ipa-user should work.
+
+        Test for : https://pagure.io/SSSD/sssd/issue/3937
+        Steps:
+        1) setup user with ssh-key and certificate stored in ipaserver
+        2) simulate p11_child timeout
+        3) try to login via ssh using private key.
+        """
+        user = 'testsshuser'
+        passwd = 'Secret123'
+        user_key = "/tmp/ssh-key{0}".format(random.randint(1, 99))
+        pem_file = tasks.create_temp_file(self.master, directory='/tmp')
+        # Create a user with a password
+        tasks.create_active_user(self.master, user, passwd)
+        tasks.kinit_admin(self.master)
+        gen_key = "ssh-keygen -N \"""\" -f {}".format(user_key)
+        tasks.run_command_as_user(self.master, user, gen_key)
+        ssh_pub_key = self.master.get_file_contents('{}.pub'.format(
+            user_key), encoding='utf-8')
+        openssl_cmd = [
+            'openssl', 'req', '-x509', '-newkey', 'rsa:2048', '-days', '365',
+            '-nodes', '-out', pem_file, '-subj', '/CN=' + user]
+        self.master.run_command(openssl_cmd)
+        cert_b64 = self.get_cert_base64(self.master, pem_file)
+        sssd_p11_child = '/usr/libexec/sssd/p11_child'
+        backup = tasks.FileBackup(self.master, sssd_p11_child)
+        try:
+            content = '#!/bin/bash\nsleep 999999'
+            # added sleep to simulate the timeout for p11_child
+            self.master.put_file_contents(sssd_p11_child, content)
+            self.master.run_command('ipa user-mod {} --ssh "{}"'.format(
+                user, ssh_pub_key))
+            self.master.run_command([
+                'ipa', 'user-add-cert', user, "--certificate=%s" % cert_b64])
```

Comment:
Please be consistent 
a) in using quotes vs double quotes - the preferred is single quotes for strings.
b) in using format styles - the preferred is using .format() method

Also I am pretty sure that `--certificate` and value of `cert_b64` can be provided as separate arguments, without `=` between them